### PR TITLE
Retrieve Personal Data

### PR DESCRIPTION
**Pull request by fenek**
This PR addresses adds CLI command to retrieve all personal data of a specific JID.

### CODE REVIEW

**Comment by fenek** [comment]
@aleklisi Is this branch up to date? Do you have any local changes that you would like to push?

**Comment by mongoose-im** [comment]
[6296.1](https://travis-ci.org/esl/MongooseIM/jobs/516053496) / Erlang 19.3 / small_tests / bda131f1e52ef6a09f0a6220e7d1ad4b2d72e01b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6296/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6296/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-05_06.58.32/lib.mongooseim.logs/run.2019-04-05_06.58.33/suite.log.html)

**Comment by mongoose-im** [comment]
[6297.1](https://travis-ci.org/esl/MongooseIM/jobs/516081131) / Erlang 19.3 / small_tests / 0bd7a06f85c2e7f40dfce1faed7d22597ecbb304
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6297/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6297/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-05_08.34.49/lib.mongooseim.logs/run.2019-04-05_08.34.50/suite.log.html)

File: src/mod_vcard_mnesia.erl [hunk 1]
```diff
@@ -15,6 +16,24 @@
...
+
+-export([get_personal_data/2]).
+
+-spec get_personal_data(gdpr:username(), gdpr:domain()) ->
+    [{gdpr:binary_table_name(), gdpr:schema(), gdpr:entities()}].
+
+get_personal_data(Username, Server) ->
+    LUser = jid:nodeprep(Username),
+    Table = vcard,
+    Schema = mnesia:table_info(Table, wild_pattern),
```
Comment (aleklisi) [inline src/mod_vcard_mnesia.erl]: Might be required to fix the schema.

**Comment by rslota** [comment]
Hi @fenek @aleklisi  ,

Is CSV format somehow enforced or is that a choice that has been made? If we made that choice, then what was the motivation?

**Comment by mongoose-im** [comment]
[6300.1](https://travis-ci.org/esl/MongooseIM/jobs/517215590) / Erlang 19.3 / small_tests / 02e7bc3f6294520f1db5a95ccc76390a42c6bdc2
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6300/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6300/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-08_11.48.29/lib.mongooseim.logs/run.2019-04-08_11.48.30/suite.log.html)

**Comment by mongoose-im** [comment]
[6301.1](https://travis-ci.org/esl/MongooseIM/jobs/517252383) / Erlang 19.3 / small_tests / 457b7af655569d7b4eeb94c7746c51e493c5db85
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6301/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6301/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-08_13.15.21/lib.mongooseim.logs/run.2019-04-08_13.15.22/suite.log.html)

**Comment by mongoose-im** [comment]
[6302.1](https://travis-ci.org/esl/MongooseIM/jobs/517662932) / Erlang 19.3 / small_tests / f59f0fffd80ab4b29c1a74fda8b38424e150371d
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6302/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6302/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-09_07.57.56/lib.mongooseim.logs/run.2019-04-09_07.57.57/suite.log.html)

**Comment by mongoose-im** [comment]
[6303.1](https://travis-ci.org/esl/MongooseIM/jobs/517686704) / Erlang 19.3 / small_tests / f02b02d3b79a68ea46130e6c0cf1ae2c6d39e94b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6303/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6303/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-09_09.06.11/lib.mongooseim.logs/run.2019-04-09_09.06.12/suite.log.html)

**Comment by mongoose-im** [comment]
[6304.1](https://travis-ci.org/esl/MongooseIM/jobs/517717750) / Erlang 19.3 / small_tests / e0474a80c8ac78c18457b57ffcec7f565cdc9628
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6304/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6304/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-09_10.27.45/lib.mongooseim.logs/run.2019-04-09_10.27.46/suite.log.html)

**Comment by mongoose-im** [comment]
[6309.1](https://travis-ci.org/esl/MongooseIM/jobs/519156107) / Erlang 19.3 / small_tests / e1d6af18b9bd71eb7118517935e3ccb6b560f72a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6309/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6309/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-12_09.04.17/lib.mongooseim.logs/run.2019-04-12_09.04.17/suite.log.html)

**Comment by mongoose-im** [comment]
[6312.1](https://travis-ci.org/esl/MongooseIM/jobs/520299639) / Erlang 19.3 / small_tests / 9d3f873eabf834b0523e2e4979dd7e86aed2a0eb
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6312/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6312/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-15_13.49.30/lib.mongooseim.logs/run.2019-04-15_13.49.31/suite.log.html)

**Comment by mongoose-im** [comment]
[6314.1](https://travis-ci.org/esl/MongooseIM/jobs/520632716) / Erlang 19.3 / small_tests / 231e99f7ee253f723cf505df7c03274541d91eda
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6314/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6314/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-16_08.26.18/lib.mongooseim.logs/run.2019-04-16_08.26.18/suite.log.html)

**Comment by mongoose-im** [comment]
[6342.1](https://travis-ci.org/esl/MongooseIM/jobs/523556954) / Erlang 19.3 / small_tests / 97b94957a538d44db132e52ffe42d9c99c22a634
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2253/6342/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2253/6342/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-23_16.35.05/lib.mongooseim.logs/run.2019-04-23_16.35.06/suite.log.html)

File: src/mod_vcard_mnesia.erl [hunk 2]
```diff
@@ -15,6 +16,25 @@
...
+%%--------------------------------------------------------------------
+%% gdpr callbacks
+%%--------------------------------------------------------------------
+
+-export([get_personal_data/2]).
+
+-spec get_personal_data(gdpr:username(), gdpr:domain()) ->
+    [{gdpr:binary_table_name(), gdpr:schema(), gdpr:entities()}].
+
+get_personal_data(Username, Server) ->
```
Comment (arcusfelis) [inline src/mod_vcard_mnesia.erl]: Why not to just use get_vcard from the main module?
Why do we need get_personal_data/2 implemented by each backend module???

**Comment by fenek** [comment]
Replaced with #2285

**Pull request closed without merging**